validator/src/cli.rs

```diff
@@ -1074,10 +1075,10 @@ pub fn app<'a>(version: &'a str, default_args: &'a DefaultArgs) -> App<'a, 'a> {
         .arg(
             Arg::with_name("rpc_pubsub_notification_threads")
                 .long("rpc-pubsub-notification-threads")
-                .requires("full_rpc_api")
```

Comment:
This is required to work around a CLAP bug where the CLI would complain that the required arg `--full-rpc-api` was not set, despite not using the `--rpc-pubsub-notification-threads` at all.

Maybe the initial author of this knew about the bug, and wrote things the way they did to work around this. Either way, I would like to have the default value set here in the CLI. Doing so would require the addition of a manual check to ensure `--full-rpc-api` was passed.

I was also debating whether we should just avoid creating a `RpcSubscriptions` if `--full-rpc-api` is not set, but it appears things are a little more tangled as `CompletedDataSetsService` (which is always created) requires a `RpcSubscriptions`, regardless of whether or not we actually have some threads to process notifications.

Reply:
I ended up adding the manual check in: https://github.com/anza-xyz/agave/pull/158/commits/9d8c4930f7b350594d588db04817953387ab06c2

I think this item is resolved, but going to leave it open for the sake of reviewers being able to see my thought process

Reply:
i think you want `Arg::default_value_if()` below instead of removing this?

Reply:
Huh, TIL. Reinstated `.requires()` in combination with `.default_value_if()` in the latest commit; removed the manual check I added as part of that too